Firebase/Core/FIRApp.m

```diff
@@ -359,6 +361,7 @@ - (FIROptions *)options {
 }
 
 - (void)setDataCollectionDefaultEnabled:(BOOL)dataCollectionDefaultEnabled {
+  self.alreadyOutputDataCollectionFlag = NO;  // Reset this flag to re-output the new value.
```

Comment:
It would be helpful to have a log here as well, "Explicitly setting the data collection default flag to %@"

Reply:
Should be wrapped in ifdef DEBUG

Reply:
Done.